lib/routes.js

```diff
@@ -71,13 +71,13 @@ const setupRoutes = function setupRoutes() {
         }
       };
 
-      socket.invalid = (filePath = '<unknown>', compiler) => {
+      socket.invalid = (filePath, compiler) => {
         if (socket.readyState === 3) {
           return;
         }
 
         const context = compiler.context || compiler.options.context || process.cwd();
-        const fileName = filePath.replace(context, '');
+        const fileName = typeof filePath === 'string' ? filePath.replace(context, '') : '<unknown>';
```

Comment:
```suggestion
        const fileName = filePath.replace && filePath.replace(context, '') || filePath;
```

Reply:
@shellscape This change would mean `fileName` might not be a string. Will the receiving end understand it in such a case? Wouldn't it be better if we enforced its type here?

Reply:
This is one of those "mystery" bits of webpack. I can't even glean why it would be sending something other than a string. When it comes to these kinds of undocumented oddities (of which there are many in webpack), I have historically found it's best to pass it down the line and let webpack do what it wants to. I've found that trying to take a guess that a value is not intended often causes more problems than it solves.